go.mod

```diff
@@ -1,9 +1,10 @@
 module github.com/celestiaorg/go-square/v2
 
 go 1.22.5
+toolchain go1.23.5
```

Comment:
why was this added? AFAIK nmt v0.23.0 doesn't use any new Go features from 1.23.x

Reply:
I think this is a problem with unsynced version of `go` and `toolchain`.

On CI we use 1.23 but `go.mod` specifies 1.22. That's why Go on CI specifies `toolchain` version. Either bump `go` to `1.23` or leave `toolchain` as it's.

BTW, not a big deal IMO and doesn't break anything.

Reply:
Fair enough, we can bump Go versions in a FLUP PR